app/javascript/store/modules/preferences/actions.js

```diff
@@ -0,0 +1,40 @@
+import preferencesApi from '../../../api/preferences';
+
+import {
+  LOAD_DEFAULT_PREFERENCES,
+  LOAD_TEAM,
+  LOAD_ORGANIZATION,
+  LOAD_TIMESPAN,
+} from '../../action-types';
+
+import {
+  SET_ORGANIZATION,
+  SET_TEAM,
+  SET_TIMESPAN,
+  SET_FETCHING_DEFAULT_PREFERENCES,
+} from '../../mutation-types';
+
+export default {
+  [LOAD_DEFAULT_PREFERENCES]({ commit }, userId) {
+    commit(SET_FETCHING_DEFAULT_PREFERENCES, true);
+
+    return preferencesApi.getPreferences(userId)
+      .then(data => {
+        const preferences = data.data.data.attributes;
+
+        commit(SET_ORGANIZATION, preferences.defaultOrg);
+        commit(SET_TEAM, preferences.defaultTeam);
+        commit(SET_TIMESPAN, preferences.defaultTime);
+      })
+      .finally(() => commit(SET_FETCHING_DEFAULT_PREFERENCES, false));
+  },
+  [LOAD_ORGANIZATION]({ commit }, organization) {
+    commit(SET_ORGANIZATION, organization);
```

Comment:
En lo personal creo que estos métodos no son necesarios. De hecho el `LOAD_ORGANIZATION` me confunde porque esperaría que la fuera a buscar pero en verdad llama al `SET_ORGANIZATION`. Me parece que se puede usar directamente `this.$store.commit(SET_ORGANIZATION, organization);` en un componente.

Reply:
de acuerdo